### PR TITLE
WIP - Conversion refactor proof of concept

### PR DESCRIPTION
**Pull request by cannawen**
## Brief description of changes:
- This is a WIP/proof of concept
- Pulled 3 conversion into their own separate file

Downside: They used a lot of shared functions, so I created a "shared" thing (final name TBD) so we can pull functions from it

TODO:  
- Create "default" values so we do not need to write "isZeroOrNegative" or "isHyperbole" all the time
- Refactor to dynamically generate a list of all regexes, instead of putting everything in a hard-coded array

Reviews welcome. Is this generally the direction we want to go in?

## Relevant link(s):

Related to: #87

### CODE REVIEW

File: src/conversions/conversion_helper.js [hunk 1]
```diff
@@ -134,49 +134,9 @@ const ukSubreddits = ["britain", "british", "england", "english", "scotland", "s
...
-  {
-    "imperialUnits" : [/f(?:oo|ee)?t (?:\/|per) s(?:ec(?:ond)?)?/],
-    "standardInputUnit" : " ft/sec",
-    "isInvalidInput" : isZeroOrNegative,
-    "isWeaklyInvalidInput" : isHyperbole,
-    "conversionFunction" : (i) => velocityMap(i.map((j) => j * 0.3048)) // 1 ft/s = 0.3048 m/s
-  },
+  require("./all/mpg"),
+  require("./all/mph"),
+  require("./all/ftPerSec"),
```
Comment (cannawen) [inline src/conversions/conversion_helper.js:139]: To be refactored so this array is generated programatically

File: src/conversions/shared.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  "volumeMap" : volumeMap,
+  "areaMap" : areaMap,
+  "pressureMap" : pressureMap,
+  "velocityMap" : velocityMap,
+  "createMap" : createMap,
+
+  "isZeroOrNegative" : isZeroOrNegative,
+  "isHyperbole" : isHyperbole,
+  "ukSubreddits" : ukSubreddits
+}
```
Comment (cannawen) [inline src/conversions/shared.js:127]: I'm not so happy about this file, but a lot of our "conversions" need access to similar functions, and we don't want to duplicate "distanceMap" logic across every single conversion... do we? Should we just drop the idea of having "dynamic" better-suited values, and always just do `miles->km` for example, and never `miles->meters`?

Comment (nalinbhardwaj) [reply src/conversions/shared.js:127]: What do you think about having deeper folders, something like `conversions/velocity/ftpersec.js` to eliminate cluttering of this file, and a separate(say `index.js` in `conversions/velocity` to contain these). `index.js` could also contain all `require(file.js)` statements, so we could just do `require(folder_name)` at top level.

Side note, I don't support the idea of completely dropping "dynamic" better-suited values myself.

Comment (cannawen) [reply src/conversions/shared.js:127]: Hmmm, I like the idea of deeper folders. I'll try playing around with what it would look like

**Pull request closed without merging**